src/Runtime/Distributions/IImmutableDistribution.cs

```diff
@@ -0,0 +1,297 @@
+        TDist WithMeanAndVariance(MeanType mean, VarType variance);
+    }
+
+    /// <summary>
+    /// Whether the distribution supports creating copies with all elements set to duplicates of the same value.
+    /// </summary>
+    /// <typeparam name="T">Type of the element.</typeparam>
+    /// <typeparam name="TDist">Type of the distribution.</typeparam>
+    [Quality(QualityBand.Experimental)]
+    public interface CanCopyWithAllElementsSetTo<in T, out TDist>
+    {
+        /// <summary>
+        /// Creates a copy of the current distribution with all elements set to duplicates of the given value.
+        /// </summary>
+        /// <param name="value">The value of all elements of the new distribution.</param>
+        /// <returns>The created distribution.</returns>
+        TDist WithAllElementsSetTo(T value);
+    }
+
+    /// <summary>
+    /// Supports computing the product of the current instance and another value.
+    /// </summary>
+    /// <typeparam name="TOther">The type of the second multiplier.</typeparam>
+    /// <typeparam name="TResult">The type of the product.</typeparam>
+    /// <remarks>Typically the type of the product would be the same as the type of the current instance.</remarks>
+    [Quality(QualityBand.Experimental)]
+    public interface CanComputeProduct<in TOther, out TResult>
+    {
+        /// <summary>
+        /// Computes the product of the current instance and <paramref name="other"/>.
+        /// </summary>
+        /// <param name="other">The second multiplier.</param>
+        /// <returns>The product.</returns>
+        TResult Product(TOther other);
+    }
+
+    /// <summary>
+    /// Supports computing the product of the current instance and another value that has
+    /// the same type as the product.
+    /// </summary>
+    /// <remarks>Typically the type of the second multiplier and the product would be
+    /// the same as the type of the current instance.</remarks>
+    [Quality(QualityBand.Experimental)]
+    public interface CanComputeProduct<T> : CanComputeProduct<T, T>
+    {
+    }
+
+    /// <summary>
+    /// Supports computing the ratio of the current instance and another value.
+    /// </summary>
+    /// <typeparam name="TDenominator">The type of the denominator.</typeparam>
+    /// <typeparam name="TResult">The type of the ratio.</typeparam>
+    /// <remarks>Typically the type of the ratio would be the same as the type of the current instance.</remarks>
+    [Quality(QualityBand.Experimental)]
+    public interface CanComputeRatio<in TDenominator, out TResult>
+    {
+        /// <summary>
+        /// Computes the product of the current instance and <paramref name="denominator"/>.
+        /// </summary>
+        /// <param name="denominator">The denominator.</param>
+        /// <param name="forceProper">If true, the result is modified to have parameters in a safe range</param>
+        TResult Ratio(TDenominator denominator, bool forceProper = false);
```

Comment:
Method names should be verbs

Reply:
It is more common to call this `Divide`.  See [TimeSpan.Divide](https://docs.microsoft.com/en-us/dotnet/api/system.timespan.divide).